src/org/daisy/dotify/formatter/impl/volume/TocSequenceEventImpl.java

```diff
@@ -145,6 +145,12 @@ public BlockSequence getBlockSequence(FormatterContext context, DefaultContext v
 						fsm.appendGroup(volumeEnd);
 					}
 				}
+				{
+					Collection<Block> volumeToc = data.filter(refToVolume(null, crh));
+					if (!volumeToc.isEmpty()) {
+						fsm.appendGroup(volumeToc);
```

Comment:
1. It seems you are appending a `volumeToc` after you have appended the `volumeEnd`. Is that correct?
2. Do we need this change? In other words, is it desirable to have the exact same behavior?

Reply:
1. This is the behavior from before. It is neither correct or incorrect because this only happens in the intermediary solutions. In the final solution volumeToc will always be empty.
2. Not really I think. But as Daniel said it is useful for checking that PR #7 had no _real_ regressions. We can always revert this change later.

Reply:
> It seems you are appending a `volumeToc` after you have appended the `volumeEnd`.

OK I'm not 100 % sure this is the behavior from before. In my example there was no `on-volume-end`.

Reply:
I wrote:
> It seems you are appending a `volumeToc` after you have appended the `volumeEnd`.
@bertfrees are you going to fix this?

Reply:
Nah, I don't think so. Since it isn't wrong, and the goal, which is to make MTM's regression test suite pass, is achieved, I think we're done. Anything else that we would change now to make the behavior more like before would make the code less understandable I think.

Reply:
Hm.. I'm not fully convinced, but I will approve.